src/MLJXGBoostInterface.jl

```diff
@@ -1,665 +1,194 @@
-    ,colsample_bytree=1
-    ,colsample_bylevel=1
-    ,lambda=1
-    ,alpha=0
-    ,tree_method="auto"
-    ,sketch_eps=0.03
-    ,scale_pos_weight=1
-    ,updater="auto"
-    ,refresh_leaf=1
-    ,process_type="default"
-    ,grow_policy="depthwise"
-    ,max_leaves=0
-    ,max_bin=256
-    ,predictor="cpu_predictor" #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type="uniform"
-    ,normalize_type="tree"
-    ,rate_drop=0.0
-    ,one_drop=0
-    ,skip_drop=0.0
-    ,feature_selector="cyclic"
-    ,top_k=0
-    ,tweedie_variance_power=1.5
-    ,objective="count:poisson"
-    ,base_score=0.5
-    ,eval_metric="rmse"
-    ,seed=0
-    ,nthread = 1)
-
-    model = XGBoostCount(
-    num_round
-    ,booster
-    ,disable_default_eval_metric
-    ,eta
-    ,gamma
-    ,max_depth
-    ,min_child_weight
-    ,max_delta_step
-    ,subsample
-    ,colsample_bytree
-    ,colsample_bylevel
-    ,lambda
-    ,alpha
-    ,tree_method
-    ,sketch_eps
-    ,scale_pos_weight
-    ,updater
-    ,refresh_leaf
-    ,process_type
-    ,grow_policy
-    ,max_leaves
-    ,max_bin
-    ,predictor #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type
-    ,normalize_type
-    ,rate_drop
-    ,one_drop
-    ,skip_drop
-    ,feature_selector
-    ,top_k
-    ,tweedie_variance_power
-    ,objective
-    ,base_score
-    ,eval_metric
-    ,seed
-    ,nthread)
-
-     message = MMI.clean!(model)
-     isempty(message) || @warn message
-
-    return model
+function importances(X, r)
+    fs = schema(X).names
+    [named_importance(fi, fs) for fi ∈ XGBoost.importance(r)]
 end
 
-function MMI.clean!(model::XGBoostCount)
-    warning = ""
-    if(!(model.objective in ["count:poisson"]))
-        warning *= "Changing objective to \"poisson\", "*
-                       "the only supported value. "
-        model.objective="poisson"
-    end
-    return warning
-end
-
-function MMI.fit(model::XGBoostCount
-             , verbosity::Int     #> must be here even if unsupported in pkg
-             , X
-             , y)
-
-    silent = verbosity > 0 ?  false : true
-
+function MMI.fit(model::XGBoostAbstractRegressor, verbosity::Integer, X, y)
     Xmatrix = _to_array(MMI.matrix(X))
-    dm = XGBoost.DMatrix(Xmatrix, label=_to_array(y))
-
-    seed =
-        model.seed == -1 ? generate_seed() : model.seed
-
-    fitresult = XGBoost.xgboost(dm, model.num_round;
-                                kwargs(model, silent, seed, "count:poisson",model.eval_metric)...)
-    features = schema(X).names
-    importances = [named_importance(fi, features) for
-                   fi in XGBoost.importance(fitresult)]
+    dm = DMatrix(Xmatrix, label=_to_array(y))
 
-    cache = nothing
-    report = (feature_importances=importances, )
+    r = xgboost(dm, model.num_round; kwargs(model, verbosity, model.objective)...)
 
-    return fitresult, cache, report
+    report = (feature_importances=importances(X, r),)
```

Comment:
Since we are refactoring, we should also use the new MLJ API for feature importance. Use [here](https://github.com/JuliaAI/MLJDecisionTreeInterface.jl/pull/28/files) as template.